src/rust/engine/process_execution/src/remote.rs

```diff
@@ -592,6 +610,73 @@ mod tests {
     );
   }
 
+  #[test]
+  fn ensure_inline_stdio_is_stored() {
+    let test_stdout = TestData::roland();
+    let test_stderr = TestData::catnip();
+
+    let mock_server = {
+      let op_name = "cat".to_owned();
+
+      mock::execution_server::TestServer::new(mock::execution_server::MockExecution::new(
+        op_name.clone(),
+        super::make_execute_request(&cat_roland_request())
+          .unwrap()
+          .1,
+        vec![
+          make_successful_operation(
+            &op_name.clone(),
+            StdoutType::Raw(test_stdout.string()),
+            StderrType::Raw(test_stderr.string()),
+            0,
+          ),
+        ],
+      ))
+    };
+
+    let store_dir = TempDir::new("store").unwrap();
+    let cas = mock::StubCAS::with_content(1024, vec![], vec![TestDirectory::containing_roland()]);
```

Comment:
I would probably use a `mock::StubCAS::empty()` here, and maybe an action which doesn't have an input directory. Otherwise it's a little unclear/confounding to work out what was added and what was already there.

Reply:
fixed!